File: Js/121_Best_Time_Buy_Sell.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = (prices) =>{
    let minPrice = prices[0];
    let maxProfit = 0;
    
    for(let i=1; i<prices.length; i++)
    {
        maxProfit = Math.max(maxProfit,prices[i] - minPrice);
        minPrice = Math.min(minPrice,prices[i]);
    }
    return maxProfit;
};

File: C#/121_Best_Time_Buy_Sell.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int maxProfit = 0, minPrice = prices[0];
        
        for(var i=1; i<prices.Length; i++)
        {
            maxProfit = Math.max(maxProfit,prices[i] - minPrice);
            minPrice = Math.min(minPrice,prices[i]);
        }
    return maxProfit;
    }
}